pkg/client/client.go

```diff
@@ -2032,3 +2001,44 @@ func deleteOptions(dp metav1.DeletionPropagation) metav1.DeleteOptions {
 		PropagationPolicy: &dp,
 	}
 }
+
+type pollOptions struct {
+	timeout   time.Duration
+	interval  time.Duration
+	lastError *error
+}
+
+func WithPollTimeout(d time.Duration) func(o *pollOptions) {
+	return func(o *pollOptions) {
+		o.timeout = d
+	}
+}
+
+func WithPollInterval(d time.Duration) func(o *pollOptions) {
+	return func(o *pollOptions) {
+		o.interval = d
+	}
+}
+
+func WithLastError(e *error) func(o *pollOptions) {
+	return func(o *pollOptions) {
+		o.lastError = e
+	}
+}
+
+// Poll is a wrapper around wait.PollUntilContextTimeout that allows adding the passed lastError into
+// the final error if if set by the condition, this would add more context to the "context deadline exceeded" error e.g..
+func Poll(ctx context.Context, condition wait.ConditionWithContextFunc, options ...func(o *pollOptions)) error {
+	opts := pollOptions{timeout: 5 * time.Minute, interval: time.Second}
+	for _, o := range options {
+		o(&opts)
+	}
+
+	if err := wait.PollUntilContextTimeout(ctx, opts.interval, opts.timeout, false, condition); err != nil {
+		if opts.lastError != nil && *opts.lastError != nil {
+			err = fmt.Errorf("%w: lastError: %w", err, *opts.lastError)
```

Comment:
(nit) I'm wondering if we need the intermediate `lastError: ` string?
```suggestion
			err = fmt.Errorf("%w: %w", err, *opts.lastError)
```